### PR TITLE
CB: Handle LeftShift idStatus and add limited type handling for Guard

### PR DESCRIPTION
**Pull request by drostron**
Also, a few fixes/updates were made to enable some related flatten tests.

Regarding `BasicQueryEnablementSpec`, due to fragility I'm pruning tests that fail due to stale N1QL when covered by query integration tests.

### CODE REVIEW

File: couchbase/src/main/scala/quasar/physical/couchbase/planner/MapFuncPlanner.scala [hunk 1]
```diff
@@ -93,7 +96,7 @@ final class MapFuncPlanner[F[_]: Monad: NameGenerator, T[_[_]]: Recursive: ShowT
     // array
     case Length(a1) =>
       val a1N1ql = n1ql(a1)
-      partialQueryString(s"ifnull(length($a1N1ql), array_length($a1N1ql))").point[M]
+      partialQueryString(s"ifnull(length($a1N1ql), array_length($a1N1ql), object_length($a1N1ql))").point[M]
```
Comment (sellout) [inline couchbase/src/main/scala/quasar/physical/couchbase/planner/MapFuncPlanner.scala:99]: I’m not sure we define length for objects. Although, some notion of size seems reasonable, so why not this one? But maybe we should rename it to `Size` then (not in this PR).

File: it/src/main/resources/tests/largestCities.test [hunk 2]
```diff
@@ -2,7 +2,8 @@
     "name": "largest cities",
     "backends": {
         "postgresql": "pending",
-        "marklogic":  "skip"
+        "marklogic":  "skip",
+        "couchbase":  "skip"
```
Comment (sellout) [inline it/src/main/resources/tests/largestCities.test:6]: Wait a second … any idea why this test now breaks?

**Review by sellout — CHANGES_REQUESTED** [review]
Just want some info on the newly-broken test.

File: it/src/main/resources/tests/largestCities.test (hunk 2 above)
Comment (drostron) [reply it/src/main/resources/tests/largestCities.test:6]: The generated N1QL is now quite large and something about it causes the test to time out. We'll need to get back to it, but I didn't want it to hold this up.

**Review by sellout — APPROVED** [review]

**Pull request merged**